files/rbac/mappings/stormers.yaml

```diff
@@ -0,0 +1,7 @@
+# StackStorm RBAC LDAP groups-to-roles mapping rules, shipped in '/opt/stackstorm/rbac/mappings/'
+# See RBAC Roles Based on LDAP Groups: https://docs.stackstorm.com/rbac.html#automatically-granting-roles-based-on-ldap-group-membership
+---
+group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
+description: "Automatically grant admin role to all stormers group members."
+roles:
+  - "admin"
```

Comment:
```suggestion
# ---
# group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
# description: "Automatically grant admin role to all stormers group members."
# roles:
#   - "admin"
```

Just one note to comment out the LDAP < > RBAC mappings as that's not the default configuration and will blow up without the LDAP auth settings in st2.conf.

Besides that, looks good. Great work!

Reply:
Yes I already tried with them all commented out, however `st2-apply-rbac-definitions` fails to execute the import when there's a file with nothing in it. I ran the sync with that file uncommented and it runs fine.

```shell
root@49f0bd852fa5:/opt/stackstorm# cat rbac/mappings/stormers.yaml 
# StackStorm RBAC LDAP groups-to-roles mapping rules, shipped in '/opt/stackstorm/rbac/mappings/'
# See RBAC Roles Based on LDAP Groups: https://docs.stackstorm.com/rbac.html#automatically-granting-roles-based-on-ldap-group-membership
#---
#group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
#description: "Automatically grant admin role to all stormers group members."
#roles:
#  - "admin"
root@49f0bd852fa5:/opt/stackstorm# st2-apply-rbac-definitions --config-file /etc/st2/st2.docker.conf
2021-04-09 22:59:41,484 INFO [-] Connecting to database "st2" @ "mongo:27017" as user "None".
2021-04-09 22:59:41,495 INFO [-] Successfully connected to database "st2" @ "mongo:27017" as user "None".
2021-04-09 22:59:41,943 INFO [-] Loading role definitions from "/opt/stackstorm/rbac/roles/"
2021-04-09 22:59:42,086 INFO [-] Loading user role assignments from "/opt/stackstorm/rbac/assignments/"
2021-04-09 22:59:42,096 INFO [-] Loading group to role map definitions from "/opt/stackstorm/rbac/mappings/"
Traceback (most recent call last):
  File "/usr/bin/st2-apply-rbac-definitions", line 15, in <module>
    sys.exit(apply_rbac_definitions.main(sys.argv[1:]))
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/cmd/apply_rbac_definitions.py", line 60, in main
    apply_definitions()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/cmd/apply_rbac_definitions.py", line 44, in apply_definitions
    result = loader.load()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/loader.py", line 65, in load
    result['group_to_role_maps'] = self.load_group_to_role_maps()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/loader.py", line 136, in load_group_to_role_maps
    file_path=file_path)
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2rbac_backend/loader.py", line 191, in load_group_to_role_map_assignment_from_file
    raise ValueError(msg)
ValueError: Group to role map assignment file "/opt/stackstorm/rbac/mappings/stormers.yaml" is empty and invalid
```

How do you want to handle this?

Reply:
Thanks for trying that!
I'm fine with removing the example mappings file then.